### PR TITLE
Extracting text from a real time captured image (#497)

### LINKED ISSUE #497: Extracting the text from a real-time captured image.(GSSOC '21)
# Aim

### Extracting the text from a real-time captured image.

# Details
By using Open CV we read a real-time image and by applying various filters we try to reduce the noise after that with the use of pytesseract we extract the text written in the image
Libraries used-
* cv2
* pytesseract

### PR DESCRIPTION
**Pull request by ShubhamGupta577**
# Description

In this script, we are going to read a real-time captured image using a webcam or any other camera and extract the text from that image by removing noise from it.
This script uses `OpenCV` for capturing the real-time image and `pytesseract` for text extraction from the captured image.
Here I added two types of files as one `.py` which can be used on a machine and the other is with the `.ipynb` extension so that it can be used on google colab which removes the need for external installation.

## Fixes #497 
Extracting the text from a real-time captured image. (GSSOC'21)

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [x] Yes
- [ ] No

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by ShubhamGupta577** [comment]
@kaustubhgupta Please review it.

**Review by kaustubhgupta — APPROVED** [review]

File: Realtime Text Extraction/Realtime Text Extraction.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+For capturing a image press 'SPACE' when web-cam is on.
+'''
+#importing required libraries
+
+import cv2
+import pytesseract 
+from PIL import Image
+
+#Put the path where you stored the tesseract.exe file in your machine
+pytesseract.pytesseract.tesseract_cmd = r"C:\Users\91769\AppData\Local\Programs\Tesseract-OCR\tesseract.exe"
```
Comment (kaustubhgupta) [inline Realtime Text Extraction/Realtime Text Extraction.py:12]: Take the input for pytesseract path

File: Realtime Text Extraction/Realtime Text Extraction.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    if k%256==27:                                                 #Press Esc for exit
+        #For Esc key
+        print("Close")
+        break
+        
+    elif k%256==32:                                               #Press Space for capture the image
+        #For Space key
+        print("Image saved")
+        #Put the path where you want to store the captured image in your machine
+        path=r'C:\Users\91769\Desktop\images'+'\img.jpg'
```
Comment (kaustubhgupta) [inline Realtime Text Extraction/Realtime Text Extraction.py:40]: take this path as input

File: Realtime Text Extraction/Realtime Text Extraction.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        cv2.imwrite(path, thresh)
+        break
+
+src= Image.open(r"C:\Users\91769\Desktop\images\img.jpg")
```
Comment (kaustubhgupta) [inline Realtime Text Extraction/Realtime Text Extraction.py:44]: this also

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Comment by ShubhamGupta577** [comment]
@kaustubhgupta Required changes have been made. Please review them.

**Review by kaustubhgupta — APPROVED** [review]

**Comment by kaustubhgupta** [comment]
@santushtisharma10 @antrikshmisri

**Review by antrikshmisri — APPROVED** [review]

**Review by avinashkranjan — APPROVED** [review]
LGTM..🚩

**Comment by avinashkranjan** [comment]
@all-contributors please add @ShubhamGupta577  for code and documentation

**Pull request merged**